src/main/java/gift/ProductController.java

```diff
@@ -49,7 +44,7 @@ public ResponseEntity<Product> updateProducts(@PathVariable Long id, @RequestBod
 
     @DeleteMapping("/{id}")
     public ResponseEntity<Product> deleteProduct(@PathVariable Long id) {
-        if (products.findById(id) != null) {
+        if (products.findById(id).isPresent()) {
             return ResponseEntity.noContent().build();
         } else {
             return ResponseEntity.notFound().build();
```

Comment:
다른 메소드들과 마찬가지로 로직을 구현하면 좀 더 읽기 쉬울 것 같네요.
중요한 것은 아니니 보시고 넘기셔도 됩니다.

```suggestion
        if (products.findById(id).isEmpty()) {
            return ResponseEntity.notFound().build();
        }
        return ResponseEntity.noContent().build();
```